ipr/ipr.py

```
@@ -239,3 +240,60 @@ def create_key_alterations(
         {k: len(v) for k, v in counts.items()},
     )
+
+
+def germline_kb_matches(
+    kb_matches: List[KbMatch], all_variants: List[IprVariant], assume_somatic: bool = True
+) -> List[KbMatch]:
+    """Filter kb_matches for matching to germline or somatic events using the 'germline' optional property.
+
+    Statements related to pharmacogenomic toxicity or cancer predisposition are only relevant if
+    the variant is present in the germline of the patient.
+    Other statements, such as diagnostic or recurrent oncogenic mutations, are only relevant as
+    somatic events in cancer.  Germline variants are excluded from these matches.
+
+    Params:
+        kb_matches: KbMatch statements to be filtered.  'variant' properties must match 'key' in all_variants.
+        all_variants: IprVariants, with a 'germline' property, that were used for kb_matches creation.
+        assume_somatic: Whether to assume somatic or germline when no 'germline' property exists in the variant.
+    Returns:
+        filtered list of kb_matches
+    """
+    ret_list = []
+    germ_alts = [alt for alt in kb_matches if alt['category'] in GERMLINE_BASE_TERMS]
+    somatic_alts = [alt for alt in kb_matches if alt not in germ_alts]
+    if germ_alts:
+        logger.info(f"checking germline status of {GERMLINE_BASE_TERMS}")
+        for alt in germ_alts:
+            var_list = [v for v in all_variants if v['key'] == alt['variant']]
+            germline_var_list = [v for v in var_list if 'germline' in v and v['germline']]
+            if germline_var_list:
+                logger.debug(
+                    f"germline kbStatementId:{alt['kbStatementId']}: {alt['kbVariant']} {alt['category']}"
+                )
+                ret_list.append(alt)
+            elif var_list:
+                logger.debug(
+                    f"Dropping somatic match to germline kbStatementId:{alt['kbStatementId']}: {alt['kbVariant']} {alt['category']}"
+                )
+            else:  # no data
```

IIRC, If we get here it means that we have a kb-match with a variant that is not in the all variants list? If so maybe it should just throw an error rather than a warning here? since that would create a bunch of other problems in the report

---

Changed this.
You were correct.  It was not reporting the lack of germline info.